src/main/java/org/gridsuite/modification/utils/ModificationUtils.java

```diff
@@ -687,8 +689,36 @@ public ReportNode modifyBranchConnectivityAttributes(ConnectablePosition<?> conn
                                                          ReportNode connectivityReports) {
         List<ReportNode> reports = new ArrayList<>();
         processConnectivityPosition(connectablePosition, connectablePositionAdder, modificationInfos, branch.getNetwork(), reports);
-        modifyConnection(modificationInfos.getTerminal1Connected(), branch, branch.getTerminal1(), reports, ThreeSides.ONE);
-        modifyConnection(modificationInfos.getTerminal2Connected(), branch, branch.getTerminal2(), reports, ThreeSides.TWO);
+
+        List<NetworkModificationException> exceptions = new ArrayList<>();
+        // Pair for Side and Pair for update and terminal
+        List<Pair<ThreeSides, Pair<AttributeModification<Boolean>, Terminal>>> sidePairs =
```

Comment:
I think this would be more precise : 
```suggestion
        List<Pair<ThreeSides, Pair<AttributeModification<Boolean>, Terminal>>> modificationsBySides =
```
(or just sideModifications)